src/sgl/device/shader.cpp

```
@@ -277,14 +277,30 @@ void SlangSession::create_session(SlangSessionBuild& build)
     // https://github.com/shader-slang/slang/issues/8166
     session_options.add(slang::CompilerOptionName::DisableWarning, std::string_view("30856"));
-    // TODO: Globally disable warning 41012.
+    // TODO: Globally disable warning E41012.
     // Example: entry point 'foo' uses additional capabilities that are not part of the specified profile 'unknown'.
     // This warning happens on CUDA because we're not properly setting the target profile (i.e. "cuda_sm_x_x").
     session_options.add(slang::CompilerOptionName::DisableWarning, std::string_view("41012"));
+    // TODO: Globally disable warning E31010.
+    // warning[E31010]: Link-time constant sized arrays are a work in progress feature, some aspects of the reflection
+    // API may not work This is a workaround for an issue in the Slang compiler:
+    session_options.add(slang::CompilerOptionName::DisableWarning, std::string_view("31010"));
+    // TODO: Globally disable warning E30051.
```

Cppcheck (2.20.0)

[style] The function 'features' is never used.

(unusedFunction)